### PR TITLE
Fix module block, add enum initializers and fix type parameters

### PR DESCRIPTION
**Pull request by azz**
#1480

### CODE REVIEW

File: tests/flow_generic/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -22,7 +22,16 @@ var X = {
     F,
     G,
     T: (a: A, b: B, c: C, d: D, e: E, f: F) => G // eslint-disable-line space-before-function-paren
-  >(method: T, scope: any, a: A, b: B, c: C, d: D, e: E, f: F): G {}
```
Comment (vjeux) [inline tests/flow_generic/__snapshots__/jsfmt.spec.js.snap]: Any idea why this suddenly breaks?

File: src/printer.js [hunk 2]
```diff
@@ -2309,7 +2294,7 @@ function genericPrintNoParens(path, options, print, args) {
     case "TSDeclareKeyword":
       return "declare"
     case "TSModuleBlock":
-      return concat(path.map(print, "body"))
+      return join(hardline, path.map(print, "body"));
```
Comment (vjeux) [inline src/printer.js]: You probably want `printStatementSequence` instead. It handles preserving empty lines and handling `;`

**Comment by azz** [comment]
Whoa, bad `pull`, will fix.

File: tests/flow_generic/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (vjeux) [inline tests/flow_generic/__snapshots__/jsfmt.spec.js.snap]: Hey, I'm sorry but this is a regression as it was fitting in one line and now it doesn't anymore. There may be a `group()` that is no longer there that needs to be brought back.

I'm very sensitive to those things as they tend to introduce a lot of changes on real codebases.

**Comment by vjeux** [comment]
Waiting for you to figure out why it's now changing flow output. Thansk!

**Comment by azz** [comment]
Having trouble getting all the tests green after moving the typeParameter into printArgumentList.

Going between this fail:

    -function return_rest_param<Args: Array<mixed>>(
    -  f: (...args: Args) => void
    -): (...args: Args) => number {
    +function return_rest_param<Args: Array<mixed>>(f: (...args: Args) => void): (
    +  ...args: Args
    +) => number {

And this one:

    -  >(method: T, scope: any, a: A, b: B, c: C, d: D, e: E, f: F): G {},
    +  >(
    +    method: T,
    +    scope: any,
    +    a: A,
    +    b: B,
    +    c: C,
    +    d: D,
    +    e: E,
    +    f: F,
    +  ): G {},
     };

I might have to move the typeParameters out of this function.

**Comment by azz** [comment]
@vjeux Should be good now 😄

**Comment by vjeux** [comment]
yayayay, thanks!

**Pull request merged**